zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -2553,6 +2557,13 @@ public void connect(final long maxMsToWaitUntilConnected, Watcher watcher)
         close();
       }
     }
+    try {
+      if (_monitor != null) {
+        _monitor.register();
```

Comment:
Umm connect() can be called multiple times right, do we want to register monitor every time? Previously it used to register only when native zkclient was instantiated.

Reply:
Good question.
Please correct me if I'm wrong, indeed "connect" could happen many times, but not necessarily the exact method. From code those are handled by method `reconnect()`, which doesn't involve monitor registration. I'm not sure if that is expected pattern to begin with.

Reply:
I just saw that you are correct according to current code. Although connect API is public method, it's currently getting called only through constructor but if we want it that way that it should be called only once then may be we should add some sanity check before registering. I'm not sure what is the harm in registration.

Reply:
Also noticed that on line 2520, does it expect that monitor is already registered and then it tries to do that operation with monitor?